cirq-core/cirq/transformers/dynamical_decoupling.py

```diff
@@ -98,114 +110,75 @@ def _is_single_qubit_gate_moment(moment: 'cirq.Moment') -> bool:
     return True
 
 
-def _is_clifford_moment(moment: 'cirq.Moment') -> bool:
-    for op in moment.operations:
-        if op.gate is not None and isinstance(op.gate, cirq.MeasurementGate):
-            return False
-        if not cirq.has_stabilizer_effect(op):
-            return False
-    return True
-
-
-def _get_clifford_pieces(circuit: 'cirq.AbstractCircuit') -> list[Tuple[int, int]]:
-    clifford_pieces: list[Tuple[int, int]] = []
-    left = 0
-    for moment_id, moment in enumerate(circuit):
-        if not _is_clifford_moment(moment):
-            clifford_pieces.append((left, moment_id))
-            left = moment_id + 1
-    if left < len(circuit):
-        clifford_pieces.append((left, len(circuit)))
-    return clifford_pieces
+def _is_clifford_op(op: 'cirq.Operation') -> bool:
+    if op.gate is not None and isinstance(op.gate, cirq.MeasurementGate):
+        return False
+    return cirq.has_stabilizer_effect(op)
 
 
 def _is_insertable_moment(moment: 'cirq.Moment', single_qubit_gate_moments_only: bool) -> bool:
     return _is_single_qubit_gate_moment(moment) or not single_qubit_gate_moments_only
```

Comment:
swap the args of `or` to avoid calling the function if not necessary
```suggestion
    return not single_qubit_gate_moments_only or _is_single_qubit_gate_moment(moment)
```